packages/lib/src/vis/shared/ZoomSelectionMesh.tsx

```diff
@@ -0,0 +1,63 @@
+import { useThree } from '@react-three/fiber';
+import type { Vector2 } from 'three';
+
+import type { ModifierKey } from '../models';
+import { useAxisSystemContext } from './AxisSystemContext';
+import SelectionMesh from './SelectionMesh';
+import SelectionRect from './SelectionRect';
+import { useMoveCameraTo } from './hooks';
+
+interface Props {
+  modifierKey?: ModifierKey;
+}
+
+function ZoomSelectionMesh(props: Props) {
+  const { modifierKey } = props;
+
+  const { dataToWorld } = useAxisSystemContext();
+  const moveCameraTo = useMoveCameraTo();
+
+  const { width, height } = useThree((state) => state.size);
+  const camera = useThree((state) => state.camera);
+
+  const onSelection = (startPoint: Vector2, endPoint: Vector2) => {
+    if (startPoint.x === endPoint.x && startPoint.y === endPoint.y) {
+      return;
+    }
+
+    const worldStartPoint = dataToWorld(startPoint);
+    const worldEndPoint = dataToWorld(endPoint);
+    const rectWidth = Math.abs(worldEndPoint.x - worldStartPoint.x);
+    const rectHeight = Math.abs(worldEndPoint.y - worldStartPoint.y);
+
+    // Change scale first so that moveCameraTo computes the updated camera bounds
+    camera.scale.set(rectWidth / width, rectHeight / height, 1);
+    camera.updateProjectionMatrix();
+    camera.updateMatrixWorld();
+
+    const midPoint = worldEndPoint
+      .clone()
+      .sub(worldStartPoint)
+      .divideScalar(2)
+      .add(worldStartPoint);
+
+    moveCameraTo(midPoint.x, midPoint.y);
+  };
+
+  return (
+    <SelectionMesh modifierKey={modifierKey} onSelection={onSelection}>
+      {(startPoint: Vector2, endPoint: Vector2, isDragging: boolean) =>
+        isDragging ? (
+          <SelectionRect
+            startPoint={startPoint}
+            endPoint={endPoint}
+            opacity={0}
+            borderColor="black"
+          />
+        ) : null
+      }
+    </SelectionMesh>
```

Comment:
`RectSelectionMesh` API was too simple for what I wanted to do. So rather than expanding it, I made use of `SelectionMesh` directly.

Reply:
As per my review's overall comment, I think with `isDragging` gone, you could forward the `SelectionRect` props. That being said, I'm wondering if `RectSelectionMesh` and `LineSelectionMesh` are really worth having at all. I'm starting to think that the API of `SelectionMesh` is clear-enough as it is thanks to `SelectionRect` and `SelectionLine`.